main.go

```diff
@@ -335,11 +345,53 @@ func main() {
 
 		diskCache := disk.New(c.Dir, int64(c.MaxSize)*1024*1024*1024, proxyCache)
 
+		var tlsConfig *tls.Config
+		if len(c.TLSCaFile) != 0 {
+			caCertPool := x509.NewCertPool()
+			caCert, err := ioutil.ReadFile(c.TLSCaFile)
+			if err != nil {
+				log.Fatalf("Error reading TLS CA File: %v", err)
+			}
+			added := caCertPool.AppendCertsFromPEM(caCert)
+			if !added {
+				log.Fatalf("Failed to add certificate to cert pool.")
+			}
+
+			readCert, err := tls.LoadX509KeyPair(
+				c.TLSCertFile,
+				c.TLSKeyFile,
+			)
+			if err != nil {
+				log.Fatalf("Error reading certificate/key pair: %v", err)
+			}
+
+			tlsConfig = &tls.Config{
+				Certificates: []tls.Certificate{readCert},
+				ClientCAs:    caCertPool,
+				ClientAuth:   tls.RequireAndVerifyClientCert,
+			}
+		} else if len(c.TLSCertFile) != 0 && len(c.TLSKeyFile) != 0 {
+			readCert, err := tls.LoadX509KeyPair(
+				c.TLSCertFile,
+				c.TLSKeyFile,
+			)
+			if err != nil {
+				log.Fatalf("Error reading certificate/key pair: %v", err)
+			}
+
+			tlsConfig = &tls.Config{
+				Certificates: []tls.Certificate{readCert},
+			}
+		} else {
+			tlsConfig = nil
```

Comment:
I don't think this else block is needed- `tlsConfig` is already nil if it has never been set.